waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -337,3 +337,54 @@
                     :interval 15
                     :timeout timeout-secs)))))
         (log/warn "skipping test as INTEGRATION_TEST_BAD_IMAGE is not specified")))))
+
+(deftest ^:parallel ^:integration-fast test-kubernetes-reverse-proxy-sidecar
+  (testing-using-waiter-url
+    (when (using-k8s? waiter-url)
+      (if (contains? (get-kubernetes-scheduler-settings waiter-url) :reverse-proxy)
+        (let [reverse-proxy-flag "REVERSE_PROXY"
```

Comment:
Nitpick: We should read this from an environment variable with a default to `REVERSE_PROXY` since the value depends on the configured predicate function.

Reply:
Since we are not storing the reverse-proxy-flag anywhere now, shouldn't this be hardcoded?

Reply:
I can reintroduce the :flag in :reverse-proxy or I can do something like this do not add any extra keys to reverse-proxy:

```
(defn envoy-sidecar-enabled?
  "Returns true if the envoy sidecar is enabled, and false otherwise"
  [_ _ {:strs [env]} _]
  (let [reverse-proxy-flag "REVERSE_PROXY"]
    (contains? env reverse-proxy-flag))
  []
  (let [reverse-proxy-flag "REVERSE_PROXY"]
    reverse-proxy-flag))
```

Reply:
We can revisit this later in another PR if needed.